.github/workflows/publish-pacts.yaml

```
@@ -0,0 +1,50 @@
+name: Publish Pacts
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ dev ]
+
+jobs:
+  setup-and-test:
+    runs-on: ubuntu-latest
+    outputs:
+      nb-b64: ${{ steps.encode.outputs.nb-b64 }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install yarn
+        run: |
+          yarn install
+
+      - name: Run tests
+        run: yarn test
+
+      - name: Encode the pact as non-breaking base 64 string
+        id: encode
+        env:
+          PACT_FULL_PATH: 'pacts/terra-ui-cbas.json'  # this is a temporary hack until WM-1858 is done.
+        run: |
+          NON_BREAKING_B64=$(cat $PACT_FULL_PATH | base64 -w 0)
+          echo "nb-b64=${NON_BREAKING_B64}" >> $GITHUB_OUTPUT
+          echo $NON_BREAKING_B64
+
+  publish-pact-workflow:
+    runs-on: ubuntu-latest
+    needs: [setup-and-test]
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - name: dispatch to terra-github-workflows
```

```suggestion
      - name: Publish Pact contracts
```

Nit: "dispatch to terra-github-workflows" isn't very descriptive about what this step is doing.